frontend/src/js/components/Store.tsx

```diff
@@ -126,12 +157,17 @@ export default class Store extends React.Component<IStoreProps, IStoreState> {
     GA.event({category: 'Tickets', action: 'Expired'});
   }
 
-  startTimer() {
-    this.timer = setTimeout(this.onTimeout.bind(this), EXPIRATION_IN_MINUTES * 60 * 1000);
-    this.setState({reservationExpirationTime: new Date(Date.now() + EXPIRATION_IN_MINUTES * 60 * 1000)});
+  startTimer(timeFrom?: string) {
+    var timeoutMilliseconds = EXPIRATION_IN_MINUTES * 60 * 1000;
+    if (timeFrom) {
+      timeoutMilliseconds -= Moment().diff(Moment.tz(timeFrom, 'Europe/Helsinki'));
```

Comment:
tämä Europe/Helsinki hyppää vähän silmille näin maailmalta katsottuna.. mitä todennäköisemmin meidän käytössä se toimii kuitenkin.